utils/json.go

```diff
@@ -102,7 +106,7 @@ func UpdateObject(old interface{}, new interface{}, option UpdateJsonOption) int
 
 			for _, oldItem := range oldValue {
```

Comment:
How about the below codes:
It tries to find the deeply equaled element first, if not found, use the item with same identifier.

```go
for _, oldItem := range oldValue {
	found := false
	for index, newItem := range newArr {
		if reflect.DeepEqual(oldItem, newItem) && !used[index] {
			res = append(res, UpdateObject(oldItem, newItem, option))
			used[index] = true
			found = true
			break
		}
	}
	if found {
		continue
	}
	for index, newItem := range newArr {
		if areSameArrayItems(oldItem, newItem) && !used[index] {
			res = append(res, UpdateObject(oldItem, newItem, option))
			used[index] = true
			break
		}
	}
}
```